frontend/taipy-gui/src/components/Taipy/Progress.spec.tsx

```diff
@@ -12,40 +12,144 @@
  */
 
 import React from "react";
-
 import { render } from "@testing-library/react";
 import "@testing-library/jest-dom";
-
-import Progress from "./Progress";
+import Progress, { getFlexDirection, getBoxWidth } from "./Progress";
 
 describe("Progress component", () => {
     it("renders circular progress without value (indeterminate)", () => {
         const { getByRole } = render(<Progress />);
         const elt = getByRole("progressbar");
         expect(elt).toHaveClass("MuiCircularProgress-root");
     });
+
     it("uses the class", async () => {
         const { getByRole } = render(<Progress className="taipy-progress" />);
         const elt = getByRole("progressbar");
         expect(elt).toHaveClass("taipy-progress");
     });
+
     it("renders circular progress with value (determinate)", () => {
         const { getByRole, getByText } = render(<Progress showValue value={50} />);
         const elt = getByRole("progressbar");
         const valueText = getByText("50%");
         expect(elt).toHaveClass("MuiCircularProgress-root");
         expect(valueText).toBeInTheDocument();
     });
-    it("renders linear progress without value (inderminate)", () => {
+
+    it("renders linear progress without value (indeterminate)", () => {
         const { getByRole } = render(<Progress linear />);
         const elt = getByRole("progressbar");
         expect(elt).toHaveClass("MuiLinearProgress-root");
     });
+
     it("renders linear progress with value (determinate)", () => {
         const { getByRole, getByText } = render(<Progress linear showValue value={50} />);
         const elt = getByRole("progressbar");
         const valueText = getByText("50%");
         expect(elt).toHaveClass("MuiLinearProgress-root");
         expect(valueText).toBeInTheDocument();
     });
+
+    it("should return null when render is false", async () => {
```

Comment:
The test label is not really relevant here.
The point would be "Should not render when...".
This 'return null' really is a technical aspect of the implementation.

Reply:
Changed to a non-technical name